### PR TITLE
satellite/metainfo: fix broken object listing

### PR DESCRIPTION
**Pull request by mniewrzal**
What: This change sets correct `EncryptedPath` for authentication validation in metainfo `ListObjects` method.

Why: Listing objects with restricted api key is broken. https://storjlabs.atlassian.net/browse/V3-2920

Please describe the tests:
 - Test 1: automatic test added
 
Please describe the performance impact: none

## Code Review Checklist (to be filled out by reviewer)
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: lib/uplink/encryption_test.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+		upCfg := &uplink.Config{}
+		upCfg.Volatile.TLS.SkipPeerCAWhitelist = true
+
+		up, err := uplink.NewUplink(ctx, upCfg)
+		require.NoError(t, err)
+		defer ctx.Check(up.Close)
+
+		encryptionAccess := scope.EncryptionAccess
+		func() {
```
Comment (ifraixedes) [inline lib/uplink/encryption_test.go]: Why a `func` and not just a scope `{}`?

File: lib/uplink/encryption_test.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+		testSatellite := planet.Satellites[0]
+		err := testUplink.CreateBucket(ctx, testSatellite, "testbucket")
+		require.NoError(t, err)
+
+		err = testUplink.Upload(ctx, testSatellite, "testbucket", "videos/status.mp4", testrand.Bytes(5*memory.KiB))
+		require.NoError(t, err)
+
+		uplinkConfig := testUplink.GetConfig(testSatellite)
+		scope, err := uplinkConfig.GetScope()
+		require.NoError(t, err)
```
Comment (ifraixedes) [inline lib/uplink/encryption_test.go]: Can you move this just above `encryptionAccess` declaration (line 42)?

File: lib/uplink/encryption_test.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+			require.NoError(t, err)
+			require.Equal(t, 1, len(list.Items))
+		}()
+
+		restrictedAPIKey, restrictedEa, err := encryptionAccess.Restrict(scope.APIKey, uplink.EncryptionRestriction{
+			Bucket:     "testbucket",
+			PathPrefix: "videos",
+		})
+		require.NoError(t, err)
+		func() {
```
Comment (ifraixedes) [inline lib/uplink/encryption_test.go]: The same question before regarding the `func` vs `{}`.

File: lib/uplink/encryption_test.go (hunk 1 above)
Comment (mniewrzal) [reply lib/uplink/encryption_test.go]: I just want to close everything before I will go to next case. I can also rename variables for project and bucket.

File: lib/uplink/encryption_test.go (hunk 3 above)
Comment (mniewrzal) [reply lib/uplink/encryption_test.go]: the same reason like in previous coment

File: lib/uplink/encryption_test.go (hunk 1 above)
Comment (ifraixedes) [reply lib/uplink/encryption_test.go]: I see, never mind, I missed the `defer` calls.

**Pull request merged**